compiler/src/lnntoamm/Types.ts

```diff
@@ -156,22 +173,61 @@ export class Builtin extends Type {
     }
   }
 
+  fieldIndices(): FieldIndices {
+    return TODO("determine if it's even worth keeping the Builtin class");
+  }
+
   instance(): Type {
     return this;
   }
 
+  isFixed(): boolean {
+    switch (this.name) {
```

Comment:
This *might* be an issue if type aliasing (eg the `type int = int64; type float = float64;` the current first stage compiler does) is still kept around.

Reply:
yeah - I haven't implemented aliasing yet so I'm not worried about it yet. I think it can be done with a `class Alias extends Type` to avoid scoping issues and just delegates to an internal delegate on every call, but I want to figure out generics first